src/wrappers/themis/react-native-themis/android/src/main/java/com/reactnativethemis/ThemisModule.java

```diff
@@ -33,608 +32,544 @@
-    } catch (ByteOverflowException e) {
-      errorCallback.invoke(e.getMessage());
-      return;
-    }
-
-    byte[] msg = message.getBytes(StandardCharsets.UTF_8);
-
-    SecureMessage secureMessage = new SecureMessage(pvtKey, pubKey);
-    try {
-      byte[] encrypted = secureMessage.wrap(msg, pubKey);
-      WritableArray response = dataSerialize(encrypted);
-      successCallback.invoke(response);
-    } catch (NullArgumentException | SecureMessageWrapException e) {
-      errorCallback.invoke(e.getMessage());
-    }
-  }
-
-  @ReactMethod
-  public void secureMessageDecrypt(ReadableArray message,
-                                   ReadableArray privateKey,
-                                   ReadableArray publicKey,
+                                Callback errorCallback) {
+
+        if (privateKey == null || privateKey.size() == 0) {
+            WritableMap error = new WritableNativeMap();
+            error.putString("message", privateKeyRequired);
+            errorCallback.invoke(error);
+            return;
+        }
+
+        try {
+            byte[] bkey = dataDeserialize(privateKey);
+            PrivateKey pvtKey = new PrivateKey(bkey);
+            byte[] msg = message.getBytes(StandardCharsets.UTF_8);
+            SecureMessage secureMessage;
+            if (publicKey == null) {
+                secureMessage = new SecureMessage(pvtKey);
+            } else {
+                bkey = dataDeserialize(publicKey);
+                PublicKey pubKey = new PublicKey(bkey);
+                secureMessage = new SecureMessage(pvtKey, pubKey);
+            }
+            byte[] signedMessage = secureMessage.sign(msg);
+            WritableArray response = dataSerialize(signedMessage);
+            successCallback.invoke(response);
+        } catch (Exception e) {
+            WritableMap error = new WritableNativeMap();
+            error.putString("message", e.toString());
+            errorCallback.invoke(error);
+        }
+    }
+
+    @ReactMethod
+    public void secureMessageVerify(ReadableArray message,
+                                    ReadableArray privateKey,
+                                    ReadableArray publicKey,
+                                         Callback successCallback,
+                                         Callback errorCallback) {
+
+        if (publicKey == null || publicKey.size() == 0) {
+            WritableMap error = new WritableNativeMap();
+            error.putString("message", publicKeyRequired);
+            errorCallback.invoke(error);
+            return;
+        }
+
+        try {
+            byte[] bkey = dataDeserialize(publicKey);
+            PublicKey pubKey = new PublicKey(bkey);
+            byte[] msg = dataDeserialize(message);
+            SecureMessage secureMessage;
+            if (privateKey == null || privateKey.size() == 0) {
+                secureMessage = new SecureMessage(pubKey);
+            } else {
+                bkey = dataDeserialize(privateKey);
```

Comment:
do we need private key here? as I remember and what our docs says - https://docs.cossacklabs.com/themis/crypto-theory/cryptosystems/secure-message/#implementation-details
for verifying we need only public key. So, if it is legacy signature of method, we can leave it but ignore private key and don't validate.

plus we don't use it anywhere below.

Reply:
Removed privateKey from verify method. Also removed a public key from the sign method. 
Fixed the tests.